ui/app/components/multi-select-dropdown.js

```diff
@@ -0,0 +1,98 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { run } from '@ember/runloop';
+
+const TAB = 9;
+const ESC = 27;
+const SPACE = 32;
+const ARROW_UP = 38;
+const ARROW_DOWN = 40;
+
+export default Component.extend({
+  classNames: ['dropdown'],
+
+  options: computed(() => []),
+  selection: computed(() => []),
+
+  onSelect() {},
+
+  isOpen: false,
+  dropdown: null,
+
+  capture(dropdown) {
+    // It's not a good idea to grab a dropdown reference like this, but it's necessary
+    // in order to invoke dropdown.actions.close in traverseList as well as
+    // dropdown.actions.reposition when the label or selection length changes.
+    this.set('dropdown', dropdown);
+  },
+
+  didReceiveAttrs() {
+    const dropdown = this.get('dropdown');
+    if (this.get('isOpen') && dropdown) {
+      run.scheduleOnce('afterRender', () => {
+        dropdown.actions.reposition();
+      });
+    }
+  },
+
+  actions: {
+    toggle({ key }) {
+      const newSelection = this.get('selection').slice();
+      if (newSelection.includes(key)) {
+        newSelection.removeObject(key);
+      } else {
+        newSelection.addObject(key);
+      }
+      this.get('onSelect')(newSelection);
+    },
+
+    openOnArrowDown(dropdown, e) {
+      this.capture(dropdown);
+
+      if (!this.get('isOpen') && e.keyCode === ARROW_DOWN) {
+        dropdown.actions.open(e);
```

Comment:
Not sure, but you may need to `bind` this to `dropdown` so `this` within `actions.open` refers to the component rather than the `actions` object - guessing here btw

Reply:
It works as is and this is how it is used in the docs.

![image](https://user-images.githubusercontent.com/174740/51340521-fcec8200-1a43-11e9-9ef2-10654687ddfe.png)

Reply:
Cool, if thats how they document it then fair enough